src/config.rs

```diff
@@ -57,6 +57,10 @@ impl Default for Config {
 }
 
 impl Config {
+    pub fn freeze(&mut self) {
```

Comment:
I'd vote against keeping this Config state.

I have serious reasons to justify it and I do have a strong vision of how config should work. After all, I spent months designing my own library. I'd like to have some opinion on that though.

In my mind there should be two separate structs:
* one that represents the builder
* one that represents built config

The reason for it is simple - moving runtime errors that can happen with Frozen / Mutable state to compilation time. Simply speaking - make invalid state impossible thanks to the type system. It has a clear advantage of "it compiles, it most probably works", but at the same time it forbids adding new sources during program operations. However, if one keeps a reference to a builder somewhere it is always possible to reuse it. From my experience, it is not a problem to do so.